src/hmem_cuda.c

```diff
@@ -106,7 +115,29 @@ static struct cuda_ops cuda_ops = {
 cudaError_t ofi_cudaMemcpy(void *dst, const void *src, size_t size,
 			   enum cudaMemcpyKind kind)
 {
-	return cuda_ops.cudaMemcpy(dst, src, size, kind);
+	cudaError_t cuda_ret;
+	CUcontext data;
+	cuda_ret = cuda_ops.cudaMemcpy(dst, src, size, kind);
+	if (cuda_ret != cudaSuccess)
+		return cuda_ret;
+
+	/* If either dst or src buffer is not allocated,
+	 * mapped by, or registered with a CUcontext, the
+	 * cuPointerGetAttribute call will return
+	 * CUDA_ERROR_INVALID_VALUE. In this case the
+	 * cudaDeviceSynchronize() needs to be called
+	 * to ensure data consistency.
+	 */
+	if (ofi_cuPointerGetAttribute(&data, \
+	CU_POINTER_ATTRIBUTE_CONTEXT, (CUdeviceptr) dst) == CUDA_SUCCESS
+	&& ofi_cuPointerGetAttribute(&data,
+	CU_POINTER_ATTRIBUTE_CONTEXT, (CUdeviceptr) src) == CUDA_SUCCESS)
```

Comment:
indentation on the if statement is wonky. Also, remove the backslash

Reply:
Will fix.

Reply:
Fixed.